monai/apps/pathology/transforms.py

```diff
@@ -0,0 +1,182 @@
+# modified from sources:
+# - Original implementation from Macenko paper in Matlab: https://github.com/mitkovetta/staining-normalization
+# - Implementation in Python: https://github.com/schaugf/HEnorm_python
+# - Link to Macenko et al., 2009 paper: http://wwwx.cs.unc.edu/~mn/sites/default/files/macenko2009.pdf
+import cupy as cp
+
+from monai.transforms.transform import Transform
+
+
+class ExtractStainsMacenko(Transform):
+    """Class to extract a target stain from an image, using the Macenko method for stain deconvolution.
+
+    Args:
+        tli: (optional) transmitted light intensity
+        alpha: (optional) tolerance for the pseudo-min and pseudo-max
+        beta: (optional) Optical Density (OD) threshold for transparent pixels
+        max_cref: (optional) reference maximum stain concentrations for Hematoxylin & Eosin (H&E)
+    """
+
+    def __init__(self, tli: float = 240, alpha: float = 1, beta: float = 0.15, max_cref: cp.ndarray = None) -> None:
+        self.tli = tli
+        self.alpha = alpha
+        self.beta = beta
+
+        self.max_cref = max_cref
+        if self.max_cref is None:
+            self.max_cref = cp.array([1.9705, 1.0308])
+
+    def _deconvolution_extract_stain(self, img: cp.ndarray) -> cp.ndarray:
+        """Perform Stain Deconvolution using the Macenko Method, and return stain matrix for the image.
+
+        Args:
+            img: RGB image to perform stain deconvolution of
+
+        Return:
+            he: H&E OD matrix for the image (first column is H, second column is E, rows are RGB values)
+        """
+        # reshape image
+        img = img.reshape((-1, 3))
+
+        # calculate optical density
+        od = -cp.log((img.astype(cp.float) + 1) / self.tli)
+
+        # remove transparent pixels
+        od_hat = od[~cp.any(od < self.beta, axis=1)]
```

Comment:
```suggestion
        absorbance_hat = absorbance[cp.any(absorbance > self.beta, axis=1)]
```
Just changing `<` to `>` to save an operation on all values (`~`).

Reply:
Oh, thank you for pointing this out! Also changing it to `cupy.all()` instead of `cupy.any()` in this case, to maintain the logic :)